### PR TITLE
Added a util function to compute CFL time constraints

### PR DESCRIPTION
**Pull request by diogodcarvalho**
It would be nice to have grid constraints in plasmapy.simulation
This is a first example, which adds the CFL condition for an electromagnetic FDTD Yee solver

### CODE REVIEW

**Comment by diogodcarvalho** [comment]
pre-commit.ci autofix

**Comment by diogodcarvalho** [comment]
pre-commit.ci autofix

**Comment by josephrhsmith** [comment]
pre-commit.ci autofix

File: src/plasmapy/simulation/resolution_constraints.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
+"""
+Grid and time resolution constraints for numerical simulations.
+
+.. attention::
+
+   |expect-api-changes|
```
Comment (namurphy) [inline src/plasmapy/simulation/resolution_constraints.py:6]: Important to include for a new module!

File: src/plasmapy/simulation/resolution_constraints.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    Calculates the limiting time-step for a finite difference time-domain
+    electromagnetic Yee solver which uses a Cartesian grid.
+
+    This limit is defined by the Courant-Friederichs-Lewy (CFL) Condition:
+
+    .. math::
+        \Delta t = \frac{1}{c\sqrt{{\sum_{i=1}^{n} \Delta x_i^{-2}}}}
+
+    where :math:`\Delta x_i` corresponds to the grid resolution along the
+    :math:`i`\ -th dimension and :math:`n` is the number of dimensions
```
Comment (namurphy) [inline src/plasmapy/simulation/resolution_constraints.py:40]:
```suggestion
    :math:`i`\ -th dimension and :math:`n` is the number of dimensions.
```

**Review by namurphy — APPROVED** [review]
Thank you for doing this, and apologies for the delay in getting this in!  I have only one minor edit, and we can get this merged momentarily.

**Pull request merged**